src/libexpr/primops.cc

```diff
@@ -3767,6 +3767,99 @@ static RegisterPrimOp primop_hashString({
     .fun = prim_hashString,
 });
 
+static void prim_convertHash(EvalState & state, const PosIdx pos, Value * * args, Value & v)
+{
+    state.forceAttrs(*args[0], pos, "while evaluating the first argument passed to builtins.convertHash");
+    auto &inputAttrs = args[0]->attrs;
+
+    Bindings::iterator iteratorHash = getAttr(state, state.symbols.create("hash"), inputAttrs, "while locating the attribute 'hash'");
+    auto hash = state.forceStringNoCtx(*iteratorHash->value, pos, "while evaluating the attribute 'hash'");
+
+    Bindings::iterator iteratorHashAlgo = inputAttrs->find(state.symbols.create("hashAlgo"));
+    std::optional<HashType> ht = std::nullopt;
+    if (iteratorHashAlgo != inputAttrs->end()) {
+        ht = parseHashType(state.forceStringNoCtx(*iteratorHashAlgo->value, pos, "while evaluating the attribute 'hashAlgo'"));
+    }
+
+    Bindings::iterator iteratorToHashFormat = getAttr(state, state.symbols.create("toHashFormat"), args[0]->attrs, "while locating the attribute 'toHashFormat'");
+    HashFormat hf = parseHashFormat(state.forceStringNoCtx(*iteratorToHashFormat->value, pos, "while evaluating the attribute 'toHashFormat'"));
+
+    v.mkString(Hash::parseAny(hash, ht).to_string(hf, hf == SRI));
+}
+
+static RegisterPrimOp primop_convertHash({
+    .name = "__convertHash",
+    .args = {"args"},
+    .doc = R"(
+      Return the specified representation of a hash string, based on the attributes presented in *args*:
+
+      The result hash is the *toHashFormat* representation of the hash *hash*.
+
+      - `hash`
```

Comment:
I was a little confused where these were the fields of `args` or not since they came right after "Return..." and "The result hash..." sentences.

Reply:
The sentence is now moved below the argument introduction.